src/sandbox.ml

```diff
@@ -291,13 +291,15 @@ let of_settings_or_detect () =
   | None -> detect ()
 
 let save_to_settings sandbox =
+  let open Promise.Syntax in
   let to_setting = function
     | Esy (_, root) -> Setting.Esy root
     | Opam (_, switch) -> Setting.Opam switch
     | Global -> Setting.Global
     | Custom template -> Setting.Custom template
   in
-  Settings.set ~section:"ocaml" Setting.t (to_setting sandbox)
+  let+ () = Settings.set ~section:"ocaml" Setting.t (to_setting sandbox) in
```

Comment:
This change does not seem to be necessary.

Reply:
I think it is necessary as we need to wait for the change to be effective before going on to the next line

Reply:
Yes you are right. In commit 98a1b6e I moved an instruction from some place `A` to some place `B`, and forgot to remove the now useless `let*` in place `A`. I though you were referring to place `B`. Anyway, I removed the useless line :)